internal/transform/scope-html_test.go

```diff
@@ -0,0 +1,123 @@
+		buf.WriteString(fmt.Sprintf(`<%s`, node.Data))
+		for _, attr := range node.Attr {
+			if attr.Namespace != "" {
+				buf.WriteString(attr.Namespace)
+				buf.WriteString(":")
+			}
+
+			buf.WriteString(" ")
+			switch attr.Type {
+			case astro.QuotedAttribute:
+				buf.WriteString(attr.Key)
+				buf.WriteString("=")
+				buf.WriteString(`"` + attr.Val + `"`)
+			case astro.EmptyAttribute:
+				buf.WriteString(attr.Key)
+			case astro.ExpressionAttribute:
+				buf.WriteString(attr.Key)
+				buf.WriteString("=")
+				buf.WriteString(`{` + strings.TrimSpace(attr.Val) + `}`)
+			case astro.SpreadAttribute:
+				buf.WriteString(`{...` + strings.TrimSpace(attr.Val) + `}`)
+			case astro.ShorthandAttribute:
+				buf.WriteString(attr.Key)
+				buf.WriteString("=")
+				buf.WriteString(`{` + strings.TrimSpace(attr.Key) + `}`)
+			case astro.TemplateLiteralAttribute:
+				buf.WriteString(attr.Key)
+				buf.WriteString("=`" + strings.TrimSpace(attr.Val) + "`")
+			}
+		}
+		buf.WriteString(fmt.Sprintf(`></%s>`, node.Data))
+	}
+	return buf.String()
+}
+
+func TestScopeHTML(t *testing.T) {
+	// note: the tests have hashes inlined because it’s easier to read
+	// note: this must be valid CSS, hence the empty "{}"
+	tests := []struct {
+		name   string
+		source string
+		want   string
+	}{
+		{
+			name:   "none",
+			source: "<div />",
+			want:   `<div class="astro-XXXXXX"></div>`,
+		},
+		{
+			name:   "quoted",
+			source: `<div class="test" />`,
+			want:   `<div class="test astro-XXXXXX"></div>`,
+		},
+		{
+			name:   "quoted trim",
+			source: `<div class="test " />`,
+			want:   `<div class="test astro-XXXXXX"></div>`,
+		},
+		{
+			name:   "expression string",
+			source: `<div class={"test"} />`,
+			want:   `<div class={"test" + " astro-XXXXXX"}></div>`,
+		},
+		{
+			name:   "expression function",
+			source: `<div class={clsx({ [test]: true })} />`,
+			want:   `<div class={clsx({ [test]: true }) + " astro-XXXXXX"}></div>`,
+		},
+		{
+			name:   "empty",
+			source: "<div class />",
+			want:   `<div class="astro-XXXXXX"></div>`,
+		},
+		{
+			name:   "template literal",
+			source: "<div class=`${value}` />",
+			want:   "<div class=`${value} astro-XXXXXX`></div>",
+		},
+		{
+			name:   "component className",
+			source: `<Component className="test" />`,
+			want:   `<Component className="test astro-XXXXXX"></Component>`,
+		},
+		{
+			name:   "component className expression",
+			source: `<Component className={"test"} />`,
+			want:   `<Component className={"test" + " astro-XXXXXX"}></Component>`,
+		},
+		{
+			name:   "component className shorthand",
+			source: "<Component {className} />",
+			want:   `<Component className={className + " astro-XXXXXX"}></Component>`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nodes, err := tycho.ParseFragment(strings.NewReader(tt.source), &tycho.Node{Type: astro.ElementNode, DataAtom: atom.Body, Data: atom.Body.String()})
```

Comment:
`ParseFragment` with the context of a `atom.Body` node will skip the `html/head/body` wrappers. Handy tip for other test suites.